tests/seed_isort_config_test.py

```diff
@@ -341,3 +341,61 @@ def test_missing_git_from_path(tmpdir):
                 main(())
             msg, = excinfo.value.args
             assert msg == 'Cannot find git. Make sure it is in your PATH'
+
+
+def test_newline_grab(tmpdir):
+    with tmpdir.as_cwd():
+        tmpcfg = tmpdir.join('.isort.cfg')
+        with open(tmpcfg.strpath, 'w', newline='') as cfg:
+            cfg.write('[settings]\r\nknown_third_party=\r\n')
+        tmpdir.join('g.py').write('import cfgv\n')
+        _make_git()
+
+        assert main(()) == 1
+
+        expected = '[settings]\r\nknown_third_party=cfgv\r\n'
+        with open(tmpcfg.strpath, newline='') as cfg:
+            assert cfg.read() == expected
+
+
+def test_newline_preserve(tmpdir):
+    with tmpdir.as_cwd():
+        tmpcfg = tmpdir.join('.isort.cfg')
+        with open(tmpcfg.strpath, 'w', newline='') as cfg:
+            cfg.write('[settings]\r\nknown_third_party=\r\n')
+        tmpdir.join('g.py').write('import cfgv\n')
+        _make_git()
+
+        assert main(()) == 1
+
+        expected = '[settings]\r\nknown_third_party=cfgv\r\n'
+        with open(tmpcfg.strpath, newline='') as cfg:
+            assert cfg.read() == expected
+
+        with open(tmpcfg.strpath, 'w', newline='') as cfg:
+            cfg.write('[settings]\nknown_third_party=\n')
+        tmpdir.join('g.py').write('import cfgv\n')
+        _make_git()
+
+        assert main(()) == 1
+
+        expected = '[settings]\nknown_third_party=cfgv\n'
+        with open(tmpcfg.strpath, newline='') as cfg:
+            assert cfg.read() == expected
```

Comment:
there's multiple tests in this test, they should be split up (or more likely eliminated) -- I don't think these other tests add anything beyond your original `grab` test